### DISCUSSION TITLE
Focus boxes are broken on my machine

### DISCUSSION THREAD

**Discussion by paveloom** [fltk-rs/fltk-rs#880] [Q&A]
Here is a comparison of focus boxes on my machine (left) and on a virtual machine (right). Both are the same systems, Fedora 34.

![focus-box-60](https://user-images.githubusercontent.com/49961859/132819808-39a156fe-165e-476c-97de-e40946076a24.png)
![focus-box-100](https://user-images.githubusercontent.com/49961859/132819811-e4ce0e32-21d2-4629-b356-c01b0e14011f.png)
![focus-box-140](https://user-images.githubusercontent.com/49961859/132819814-2d1d05af-c4ea-472d-8c4d-5cf7b619a471.png)
![focus-box-180](https://user-images.githubusercontent.com/49961859/132819817-a01a64a9-be26-4b47-a1cd-ba292bac0448.png)

So, somehow the focus boxes are not displaying correctly on my machine. Any ideas what's the problem here can be?

**Comment by MoAlyousef** [comment]
Hi
What’s the output of `echo $XDG_SESSION_TYPE` on both systems?

Are the graphic/compositor settings similar?

You might prefer to ask on the fltk general mailing list:
https://groups.google.com/g/fltkgeneral

Where you might also get more/better help.

**Reply by paveloom** [reply]
Hello there,

> What’s the output of `echo $XDG_SESSION_TYPE` on both systems?

Wayland. Same for X.Org though.

> Are the graphic/compositor settings similar?

Well, the virtual machine uses LLVMpipe:

![llvmpipe](https://user-images.githubusercontent.com/49961859/132861367-e1d95d2c-1542-4eba-b18c-de579cf4fc65.png)

And on the host I have Mesa Intel Graphics:

![intel](https://user-images.githubusercontent.com/49961859/132861572-8599426c-5f1d-4941-8b25-49944429b1de.png)

> You might prefer to ask on the fltk general mailing list:
> [groups.google.com/g/fltkgeneral](https://groups.google.com/g/fltkgeneral)

Thanks, will try my luck there, too.

**Reply by paveloom** [reply]
Here's the [discussion](https://groups.google.com/g/fltkgeneral/c/IrxOpdTnFx4) on the mailing list.
